VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -32,101 +33,129 @@
         return ParseHandleHelper.getActivityClass(payload);
     }
 
-    protected android.app.Notification getNotification(android.content.Context context, android.content.Intent intent) {
+    protected android.app.Notification getNotification(Context context, Intent intent) {
         try {
-
             CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
             Class activity = ParseHandleHelper.getActivityClass(payload);
-            if( activity == CourseDetailTabActivity.class ){
-                String titleTemplate =  ResourceUtil.getResourceString(R.string.COURSE_ANNOUNCEMENT_NOTIFICATION_TITLE);
+            if (activity == CourseDetailTabActivity.class) {
+                String titleTemplate = ResourceUtil.getResourceString(R.string.COURSE_ANNOUNCEMENT_NOTIFICATION_TITLE);
 
                 EventBus.getDefault().postSticky(new CourseAnnouncementEvent(
                         CourseAnnouncementEvent.EventType.MESSAGE_RECEIVED, payload.getCourseId()));
 
-                Bundle courseBundle = new Bundle();
-                courseBundle.putBoolean(Router.EXTRA_ANNOUNCEMENTS, true);
-                courseBundle.putString(Router.EXTRA_COURSE_ID, payload.getCourseId());
-
-
-                Intent resultIntent = new Intent(context, CourseDetailTabActivity.class);
-                resultIntent.setAction(Long.toString(System.currentTimeMillis()));
-                //if user launch the app from recent list, activity will still get this intent.
-                //this is one way to avoid it.
-                resultIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK |Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
-                resultIntent.putExtra( Router.EXTRA_BUNDLE, courseBundle);
-
-                // Because clicking the notification opens a new ("special") activity, there's
-                // no need to create an artificial back stack.
-                PendingIntent resultPendingIntent =  PendingIntent.getActivity(
-                        context,
-                        0,
-                        resultIntent,
-                        PendingIntent.FLAG_UPDATE_CURRENT
-                );
-
-                NotificationManager notificationManager =
-                        (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
-
                 NotificationCompat.Builder builder = new NotificationCompat.Builder(context);
                 builder.setContentTitle(titleTemplate);
                 builder.setContentText(payload.getCourseName());
                 builder.setSmallIcon(R.drawable.app_icon);
-                builder.setContentIntent(resultPendingIntent);
                 builder.setAutoCancel(true);
 
-                Uri uri= RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
+                Uri uri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
                 builder.setSound(uri);
```

Comment:
This can also be made more generic by moving it inside the `setNotificationIntents()` method, although it's name should then need be changed to something more generic.